/wrkdirs/usr/ports/x11/hyprpaper/work/hyprpaper-f77a664/src/helpers/Jpeg.cpp:8:2: error: #error "your system is not little endian, jpeg will not work, ping vaxry or something"

While building on FreeBSD/powerpc64:
```
/usr/local/bin/g++12 -DGIT_BRANCH=\"main\" -DGIT_COMMIT_HASH=\"f77a664\" -DGIT_COMMIT_MESSAGE="\"<unknown>\"" -DGIT_DIRTY=\"portbld\" -I/wrkdirs/usr/ports/x11/hyprpaper/work/hyprpaper-f77a664/. -isystem /usr/local/include/libepoll-shim -isystem /usr/local/include/cairo -isystem /usr/local/include/glib-2.0 -isystem /usr/local/lib/glib-2.0/include -isystem /usr/local/include/pixman-1 -isystem /usr/local/include/freetype2 -isystem /usr/local/include/libpng16 -O2 -pipe  -fstack-protector-strong -Wl,-rpath=/usr/local/lib/gcc12  -Wl,-rpath=/usr/local/lib/gcc12 -O2 -pipe  -fstack-protector-strong -Wl,-rpath=/usr/local/lib/gcc12  -Wl,-rpath=/usr/local/lib/gcc12  -DNDEBUG -std=c++23 -DWLR_USE_UNSTABLE -Wall -Wextra -Wno-unused-parameter -Wno-unused-value -Wno-missing-field-initializers -Wno-narrowing -D_THREAD_SAFE -pthread -MD -MT CMakeFiles/hyprpaper.dir/src/helpers/Jpeg.cpp.o -MF CMakeFiles/hyprpaper.dir/src/helpers/Jpeg.cpp.o.d -o CMakeFiles/hyprpaper.dir/src/helpers/Jpeg.cpp.o -c /wrkdirs/usr/ports/x11/hyprpaper/work/hyprpaper-f77a664/src/helpers/Jpeg.cpp
/wrkdirs/usr/ports/x11/hyprpaper/work/hyprpaper-f77a664/src/helpers/Jpeg.cpp:8:2: error: #error "your system is not little endian, jpeg will not work, ping vaxry or something"
    8 | #error "your system is not little endian, jpeg will not work, ping vaxry or something"
      |  ^~~~~
```

This is indeed big-endian system :)

@vaxerski

--- a/src/helpers/Jpeg.cpp
+++ b/src/helpers/Jpeg.cpp
@@ -4,10 +4,6 @@
 #include <sys/types.h>
 #include <unistd.h>
 
-#if __BYTE_ORDER__ != __ORDER_LITTLE_ENDIAN__
-#error "your system is not little endian, jpeg will not work, ping vaxry or something"
-#endif
-
 cairo_surface_t* JPEG::createSurfaceFromJPEG(const std::string& path) {
 
     if (!std::filesystem::exists(path)) {
@@ -44,7 +40,11 @@ cairo_surface_t* JPEG::createSurfaceFromJPEG(const std::string& path) {
     jpeg_mem_src(&decompressStruct, (const unsigned char*)imageRawData, fileInfo.st_size);
     jpeg_read_header(&decompressStruct, true);
 
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
     decompressStruct.out_color_space = JCS_EXT_BGRA;
+#else
+    decompressStruct.out_color_space = JCS_EXT_ARGB;
+#endif
 
     // decompress
     jpeg_start_decompress(&decompressStruct);
@@ -69,7 +69,7 @@ cairo_surface_t* JPEG::createSurfaceFromJPEG(const std::string& path) {
     cairo_surface_mark_dirty(cairoSurface);
     cairo_surface_set_mime_data(cairoSurface, CAIRO_MIME_TYPE_JPEG, (const unsigned char*)imageRawData, fileInfo.st_size, free, imageRawData);
     jpeg_finish_decompress(&decompressStruct);
-    jpeg_destroy_decompress(&decompressStruct);    
+    jpeg_destroy_decompress(&decompressStruct);
 
     return cairoSurface;
 }

--- a/src/helpers/Webp.cpp
+++ b/src/helpers/Webp.cpp
@@ -52,11 +52,11 @@ cairo_surface_t* WEBP::createSurfaceFromWEBP(const std::string& path) {
     }
 
 
-    if (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
-        config.output.colorspace = MODE_bgrA;
-    else
-        config.output.colorspace = MODE_Argb;
-
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    config.output.colorspace = MODE_bgrA;
+#else
+    config.output.colorspace = MODE_Argb;
+#endif
 
     const auto CAIRODATA = cairo_image_surface_get_data(cairoSurface);
     const auto CAIROSTRIDE = cairo_image_surface_get_stride(cairoSurface);

